### DISCUSSION TITLE
Import from Tslascope

### DISCUSSION THREAD

**Discussion by baabaadeftly** [teslamate-org/teslamate#1680] [Ideas]
Thanks for a great project.

Is it possible to consider adding an import function from Teslascope?

It's a great service, but I'd like to go all-in with Teslamate. Only thing holding me back is bringing in my historical data from Teslascope.

Happy to provide sample files to enable this.

**Comment by adriankumpf** [comment]
Sample files would be helpful to see how much development time such an import would require.

Perhaps someone wants to contribute the feature.

**Reply by reddit-tennis** [reply]
Stumbled upon this old discussion from Google, as I am in the same situation. After searching around I can see no one has done this yet.

Happy to help to get this moving, unfortunately I am not familiar with Elixir so I was thinking of making a Python script, but I need to know the field mapping and data formats.

Attached to this comment are some sample files pruned from my Teslascope data, I have removed coordinates and some names, but otherwise its all exact

[charges-Tesla-sample.csv](https://github.com/user-attachments/files/19097531/charges-Tesla-sample.csv)
[drive-progress-Tesla-sample.csv](https://github.com/user-attachments/files/19097533/drive-progress-Tesla-sample.csv)
[charge-progress-Tesla-sample.csv](https://github.com/user-attachments/files/19097534/charge-progress-Tesla-sample.csv)
[drives-Tesla-sample.csv](https://github.com/user-attachments/files/19097535/drives-Tesla-sample.csv)

**Reply by DrMichael** [reply]
> Happy to help to get this moving, unfortunately I am not familiar with Elixir so I was thinking of making a Python script, but I need to know the field mapping and data formats.

The attached ER diagram might help. (@JakobLichterfeld have we that somewhere in the documentation?)
![teslamate - public](https://github.com/user-attachments/assets/39829228-93b8-44f2-ba0b-6bf1387c2aee)

**Reply by JakobLichterfeld** [reply]
> (@JakobLichterfeld have we that somewhere in the documentation?)

Not that I'm aware of. Perhaps the contributions section would be a good place.



> Happy to help to get this moving, unfortunately I am not familiar with Elixir so I was thinking of making a Python script, but I need to know the field mapping and data formats.

TeslaFi Import works via csv import, we do have a script to convert to correct format, currently beeing updated: https://github.com/teslamate-org/teslamate/pull/4575

**Reply by DrMichael** [reply]
> > (@JakobLichterfeld have we that somewhere in the documentation?)
> 
> Not that I'm aware of. Perhaps the contributions section would be a good place.

I updated the ERD with the current status. Could someone please put it into https://github.com/teslamate-org/teslamate/blob/master/website/docs/development.md? I dont't know, how to put an image in there.

**Reply by JakobLichterfeld** [reply]
> Iupdated the ERD with the current status.

Ty

> Could someone please put it into

For sure, will do.

**Reply by JakobLichterfeld** [reply]
-> #4586
